### PR TITLE
feat(client): register account page

### PR DESCRIPTION
**Pull request by sujvn**
Added modal on login page where user can request an account
Added approve property to user model
Login endpoints check is user is approved 

Testing on the netlify develop link won't work correctly because we made modifications to the backend which aren't in develop yet.

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-mh21t8 Request account page](https://app.clickup.com/t/mh21t8)

File: client/src/components/RegisterUserModal.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          'The request has been sent, you will receive an email when the account has been approved',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+        onCloseComplete: onClose,
+      });
+    } catch {
+      toast({
+        title: 'Error',
+        description: 'Email or username already exists',
```
Comment (Mackbellemore) [inline client/src/components/RegisterUserModal.jsx:62]: should we just show the error response from the server here?

Comment (gordonpn) [reply client/src/components/RegisterUserModal.jsx:62]: We tried it and it only returns "Bad request status 400" or some shit.

I think it's the makeRequest is abstracting too much so I'll refactor that with Sujan on the next ticket.

Comment (Mackbellemore) [reply client/src/components/RegisterUserModal.jsx:62]: ohh i see whats happening ya

Comment (Mackbellemore) [reply client/src/components/RegisterUserModal.jsx:62]: but ya no worries

Comment (gordonpn) [reply client/src/components/RegisterUserModal.jsx:62]: > its response.data that you need

Oh yeah you're right oops

Comment (Mackbellemore) [reply client/src/components/RegisterUserModal.jsx:62]: or like error.response.data or something like that

File: client/src/components/RegisterUserModal.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import React, { Fragment, useRef, useState } from 'react';
+import styled from '@emotion/styled';
+import { makeRequest } from '../utils/api.js';
+
+const ValidationText = styled(Text)`
+  font-size: 10px;
+  padding-left: 12px;
+`;
+
+const rolesAvailable = ['Admin', 'General'];
```
Comment (Mackbellemore) [inline client/src/components/RegisterUserModal.jsx]: I would just move this to a constants file for now just so we can use it later on for the next PRs. All good from me after this

Comment (gordonpn) [reply client/src/components/RegisterUserModal.jsx]: Very sound advice

File: client/src/components/RegisterUserModal.jsx (hunk 1 above)
Comment (gordonpn) [reply client/src/components/RegisterUserModal.jsx:62]: image

Is this good?

Comment (gordonpn) [reply client/src/components/RegisterUserModal.jsx:62]: I feel like it gives too much information to the user, but at the same time it's not a *real* app

Comment (Mackbellemore) [reply client/src/components/RegisterUserModal.jsx:62]: nah lol I need to fix the error validation for mongoose. Its ok keep the generic error message for now

**Review by Mackbellemore — APPROVED** [review]

**Review by DavidLiang01 — APPROVED** [review]

**Pull request merged**